src/meta/src/controller/catalog/alter_op.rs

```diff
@@ -782,4 +784,63 @@ impl CatalogController {
 
         Ok(version)
     }
+
+    // drop table associated source is a special case of drop relation, which just remove the source object and associated state table, keeping the streaming job and fragments.
+    pub async fn drop_table_associated_source(
+        txn: &DatabaseTransaction,
+        drop_table_connector_ctx: &DropTableConnectorContext,
+    ) -> MetaResult<(Vec<PbUserInfo>, Vec<PartialObject>)> {
+        let to_drop_source_objects: Vec<PartialObject> = Object::find()
+            .filter(object::Column::Oid.is_in(vec![drop_table_connector_ctx.to_remove_source_id]))
+            .into_partial_model()
+            .all(txn)
+            .await?;
+        let to_drop_internal_table_objs: Vec<PartialObject> = Object::find()
+            .select_only()
+            .columns([
+                object::Column::Oid,
+                object::Column::ObjType,
+                object::Column::SchemaId,
+                object::Column::DatabaseId,
+            ])
+            .filter(
+                object::Column::Oid.is_in(vec![drop_table_connector_ctx.to_remove_state_table_id]),
+            )
+            .into_partial_model()
+            .all(txn)
+            .await?;
+        let to_drop_objects = to_drop_source_objects
+            .into_iter()
+            .chain(to_drop_internal_table_objs.into_iter())
+            .collect_vec();
```

Comment:
A little confused about the logic here. Why do we need to select these objects? It seems we just need the `oid`s, which is passed as arguments..

Reply:
Because we are building requests to drop the items in frontend's schema, which requires `ObjType, SchemaId, DatabaseId`. 

https://github.com/risingwavelabs/risingwave/blob/51c4809c6fdefe0595989c8cb957724e858ad01c/src/meta/src/controller/utils.rs#L1134-L1160

Reply:
Oh, I didn't notice it was returned.